### DISCUSSION TITLE
Functional and Non-functional requirements

### DISCUSSION THREAD

**Discussion by MaverickGDN03** [dylankapnias-uni/miniproject-2023-grp5#5] [General]
### Functional requirements:
Link to latex doc: https://www.overleaf.com/1449781866djqdsjkhyxpb

1. Login Page

-  Users should be able to create their account with a username and password.
- Users should be able to log in to their account using their email address.
- When creating an account, users must be able to set up their profile information.
-  Users should have a password and username recovery option.

2. Homepage

-  Users should be able to view their potential matches in a feed format. The feed will include posts/pictures, basic information and interests of the person.
- When a user is liked by another user, the other user will have priority on the current users feed.
- When two users swipe up on each other, they will get a notification.
- When two users match, a chat will be created in messages.
- Users should be able to swipe right or left to indicate whether they like or dislike a profile.
- View a persons profile in further detail when you click on it in the home page. 
- Users should be able to search for topics/interests 
-When users search for a topic/interest, their feed will be altered to show other users who have that specific feed/interest in their bio.

3. Messages

- When the allocated time runs out, users will not be able to send messages.
- Users should be able to report or block any profile that violates the application's terms and conditions or is inappropriate.
- When two users match, their chat will be allocated time which is not taken from either of their total time.
- When the message time runs out, the user who liked the other user first will be prompted to either spend their time to continue talking or to leave the chat.
- Users will get tier promotions based on how long they chat for (e.g. 30 mins promotes them to a bronze tier).
- The time prompt will alternate between the two users.
- Users will have a limited amount of days to continue talking to their match.
- The system will dematch users after a period of inactivity
- Users should be able to view their chat history.
- The user should be able to send media and text via message to another user.
- User should get a automatic message such as "You have matched with x" when their profile is liked. 
- The messaging system should indicate when a user is typing a message 
- The messaging system should indicate when a message has been read.
- The messaging system should be secure, protect user privacy by encrypting messages and preventing unauthorized access.
- Users will not be able to see the content of messages unless they are in the chat.
- Time will decrease when either or both of the two users are in the chat and typing.

4. Posts
  
- Users should be able to upload photos.
- Users should be able to remove their own posts if they no longer want them to be visible on the application.

5. Notifications
- When two users have matched they should both get a notification stating they have matched.
- Users should receive notifications when they get a message.
- Users should get a notification when they receive their daily free time allowance.
- Users will not be able to see the content of the new message, only that they have a new message from user x.

6. Profile

- Users must be able to view their profile and images.
- Users should be able to deactivate or delete their profiles if they no longer want to use the application.
- Users should be able to view their remaining time.   

7.  Settings

- User should be able to edit their account settings.
- Users should be able toe edit their profile settings.
- Users should be able to by more time.
- Users should be able to change their privacy settings.
- Users should be able to change the information displayed in their "about" page.
- Uses should have the option to log out of the application.

Non- functional requirements:

1. Performance:

- The system should be designed to have a high volume of users.
- The system should not slow down with an increase in the number of users. The system response time should close to instant.
- The system should be able to scale up or down based on user demand or application usage patterns .

2. Security: 

- The system should validate the details of a user upon account creation.
- The system should protect user data and prevent unauthorized access.

Software Quality Attributes:

1. Usability:
- The system should be easy to use and understand by any user, with intuitive user interfaces.

2. Availability: 

- System should always be available, with minimum downtime.
- System should be reliable, it should yield correct results upon user request.

4. Maintainability:

- The system should be developed in such a way that it is easy to add remove or change features.

**Comment by ogb-Welsh** [comment]
Where do we edit user profile? IN the profile page or settings page?

**Reply by MaverickGDN03** [reply]
In settings page

**Comment by MaverickGDN03** [comment]
What order is our homepage going to be in ? Location based first or category based

**Reply by MaverickGDN03** [reply]
Location based and loose matching regarding gender/sexuality etc 

Categories used more for search based filtering

**Reply by MaverickGDN03** [reply]
Only location based match making
